tests/testthat/test-summary_rows.R

```diff
@@ -819,6 +827,113 @@ test_that("`groups = FALSE` returns data unchanged", {
+    gt(rowname_col = "id", groupname_col = "group") %>%
+    summary_rows(groups = TRUE, columns = vars(value), fns = list("sum"))
+
+  gt_tbl_3 <-
+    tbl_3 %>%
+    gt(rowname_col = "id", groupname_col = "group") %>%
+    summary_rows(groups = TRUE, columns = vars(value), fns = list("sum"))
+
+  # Expect the correct values in summary rows of `gt_tbl`
+  gt_tbl_1 %>% render_as_html() %>% xml2::read_html() %>%
+    selection_text("[class='gt_row gt_right gt_summary_row gt_first_summary_row']") %>%
+    expect_equal(c("3.00", "20.00", "99.00"))
+
+  # Expect the HTML output tables of `gt_tbl_1` and `gt_tbl_1b` to be the same
+  expect_identical(
+    gt_tbl_1 %>% render_as_html(),
+    gt_tbl_1b %>% render_as_html()
+  )
+
+  # Expect the correct values in summary rows of `gt_tbl_2`
+  gt_tbl_2 %>% render_as_html() %>% xml2::read_html() %>%
+    selection_text("[class='gt_row gt_right gt_summary_row gt_first_summary_row']") %>%
+    expect_equal(c("3.00", "99.00", "20.00"))
+
+  # Expect the correct values in summary rows of `gt_tbl_3`
+  gt_tbl_3 %>% render_as_html() %>% xml2::read_html() %>%
+    selection_text("[class='gt_row gt_right gt_summary_row gt_first_summary_row']") %>%
+    expect_equal(c("20.00", "3.00", "99.00"))
+
+  # Prepare a set gt tables with a grand summary (using the same
+  # `grand_summary_rows()` call each time)
+  gt_tbl_1_gs <-
+    tbl_1 %>%
+    dplyr::group_by(group) %>%
+    gt(rowname_col = "id") %>%
+    grand_summary_rows(columns = vars(value), fns = list("sum"))
+
+  gt_tbl_1b_gs <-
+    tbl_1 %>%
+    gt(rowname_col = "id", groupname_col = "group") %>%
+    grand_summary_rows(columns = vars(value), fns = list("sum"))
+
+  gt_tbl_2_gs <-
+    tbl_2 %>%
+    gt(rowname_col = "id", groupname_col = "group") %>%
+    grand_summary_rows(columns = vars(value), fns = list("sum"))
+
+  gt_tbl_3_gs <-
+    tbl_3 %>%
+    gt(rowname_col = "id", groupname_col = "group") %>%
+    grand_summary_rows(columns = vars(value), fns = list("sum"))
+
+  # Expect the correct value in the grand summary row of `gt_tbl_gs`
+  gt_tbl_1_gs %>% render_as_html() %>% xml2::read_html() %>%
+    selection_text("[class='gt_row gt_right gt_grand_summary_row gt_first_grand_summary_row']") %>%
+    expect_equal(c("122.00"))
+
+  # Expect the HTML output tables of `gt_tbl_gs` and `gt_tbl_1b_gs` to be the same
+  expect_identical(
+    gt_tbl_1_gs %>% render_as_html(),
+    gt_tbl_1b_gs %>% render_as_html()
+  )
+
+  # Expect the correct value in the grand summary row of `gt_tbl_2_gs`
+  gt_tbl_2_gs %>% render_as_html() %>% xml2::read_html() %>%
+    selection_text("[class='gt_row gt_right gt_grand_summary_row gt_first_grand_summary_row']") %>%
+    expect_equal(c("122.00"))
+
+  # Expect the correct value in the grand summary row of `gt_tbl_3_gs`
+  gt_tbl_3_gs %>% render_as_html() %>% xml2::read_html() %>%
+    selection_text("[class='gt_row gt_right gt_grand_summary_row gt_first_grand_summary_row']") %>%
+    expect_equal(c("122.00"))
+})
+
```

Comment:
Let's add an example where the original data frame has the column name `columns`.

Reply:
Now added.